### ISSUE TITLE
False positive F401

### ISSUE THREAD

**Issue by yukihiko-shinoda** [PyCQA/flake8#2027]
### how did you install flake8?

```console
uv sync
```

### unmodified output of `flake8 --bug-report`

```json
{
  "platform": {
    "python_implementation": "CPython",
    "python_version": "3.14.5",
    "system": "Linux"
  },
  "plugins": [
    {
      "plugin": "Flake8-pyproject",
      "version": "1.2.4"
    },
    {
      "plugin": "cohesion",
      "version": "1.2.0"
    },
    {
      "plugin": "dlint",
      "version": "0.16.0"
    },
    {
      "plugin": "flake8-bugbear",
      "version": "25.11.29"
    },
    {
      "plugin": "hacking",
      "version": "8.0.0"
    },
    {
      "plugin": "mccabe",
      "version": "0.7.0"
    },
    {
      "plugin": "pycodestyle",
      "version": "2.14.0"
    },
    {
      "plugin": "pyflakes",
      "version": "3.4.0"
    }
  ],
  "version": "7.3.0"
}
```

#### what I expected to happen

No warnings

#### sample code

```python
from datetime import date
from typing import TypedDict


class Test(TypedDict):
    data_id: int
    date: date
```

#### commands ran

```console
$ # uv run flake8 test.py
test.py:1:1: F401 'datetime.date' imported but unused
from datetime import date
^
1     F401 'datetime.date' imported but unused
```

**Comment by asottile** [comment]
please check the issue template (it tells you where to search for your particular error and to find the relevant duplicate which explains why this is in fact not a false positive)
